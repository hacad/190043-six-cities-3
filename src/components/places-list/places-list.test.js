import React from "react";
import renderer from "react-test-renderer";
import PlacesList from "./places-list.jsx";

const cityPlaces = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    type: `Apartment`,
    img: `img/apartment-01.jpg`,
    category: `Premium`,
    name: `Beautiful &amp; luxurious apartment at great location`,
    price: {
      value: 120,
      currency: `â‚¬`
    },
    rating: 93,
    bookmarked: false,
    location: {
      latitude: 52.370216,
      longitude: 4.895168
    }
  }
];

it(`PlacesList correctly renders after relaunch`, () => {
  const placesList = renderer
    .create(<PlacesList
      cityPlaces={cityPlaces}
      onClickCardHeader={jest.fn()}/>)
    .toJSON();

  expect(placesList).toMatchSnapshot();
});
