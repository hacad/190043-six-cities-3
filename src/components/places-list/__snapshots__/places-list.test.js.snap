// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesList correctly renders after relaunch 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <<div />
    onActivate={[MockFunction]}
    onClickHeader={[MockFunction]}
    onDeactivate={[MockFunction]}
    place={
      Object {
        "city": Object {
          "location": Object {
            "latitude": 52.370216,
            "longitude": 4.895168,
            "zoom": 10,
          },
          "name": "Amsterdam",
        },
        "id": 1,
        "isFavorite": false,
        "isPremium": false,
        "location": Object {
          "latitude": 52.370216,
          "longitude": 4.895168,
        },
        "previewImage": "img/apartment-01.jpg",
        "price": 120,
        "rating": 93,
        "title": "Beautiful &amp; luxurious apartment at great location",
        "type": "apartment",
      }
    }
  />
</div>
`;
